### DISCUSSION TITLE
Do not list or deprioritize magic-methods (e.g., `__gt__`) in completion list

### DISCUSSION THREAD

**Discussion by goekce** [hrsh7th/nvim-cmp#1718] [Q&A] [answered]
If I autocomplete the symbols of a `list` in Python, then I get a long list. What I rather want is a shorter list of public methods as autocompleted in `ipython`.

Should I configure this in the language server (e.g., `ruff`) or can `nvim-cmp` ignore/deprioritize results based on the symbol prefix (e.g., `__*`)?

**Comment by fitrh** [comment, marked as answer]
You can write what we call `comparator`, see [lua/cmp/config/compare.lua](https://github.com/hrsh7th/nvim-cmp/blob/main/lua/cmp/config/compare.lua) for an example.

For your use case, I think you can use [lukas-reineke/cmp-under-comparator](https://github.com/lukas-reineke/cmp-under-comparator).
